packages/ui/src/components/NumberInput/index.tsx

```diff
@@ -291,6 +278,7 @@ export const NumberInput = ({
         }}
         aria-live="assertive"
         role="status"
+        data-disabled={disabled}
```

Comment:
Optional too :
Do we need to duplicate the prop instead of something like 

const StyledContainer = styled`
   &[data-disabled=true] ${StyledCenterBox} {
       border: none;
   }
`